src/core/util/schemasUtil.ts

```diff
@@ -0,0 +1,266 @@
+    // some mappings are 'reversed'
+    if (typeof key === 'object') {
+      const result = doFindByUUID(key, uuid);
+      if (result) {
+        return result;
+      }
+    }
+  }
+  return undefined;
+};
+
+export const calculatePath = (
+  root: any,
+  object: any
+): Array<string> | CalculatePathError => {
+  const path = doCalculatePath(root, object);
+  if (!path) {
+    return {
+      id: 'calulatePathError',
+      root: root,
+      element: object,
+    };
+  }
+  return path;
+};
+
+export const getPathString = (object: any): string | PathError => {
+  const root = getRoot(object);
+  const path = calculatePath(root, object);
+  if (isPathError(path)) {
+    return path;
+  }
+  return `${path.join('/')}`;
+};
+
+const doCalculatePath = (root: any, object: any): Array<string> | undefined => {
+  if (root === object) {
+    return [];
+  }
+  const entries = root instanceof Map ? root.entries() : Object.entries(root);
+  for (const [key, value] of Array.from(entries)) {
+    if (value === object) {
+      return [key];
+    }
+    // some mappings are 'reversed'
+    if (key === object) {
+      return [value];
+    }
+    if (typeof value === 'object' && key !== 'parent') {
+      const path = doCalculatePath(value, object);
+      if (path) {
+        return [key, ...path];
+      }
+    }
+    // some mappings are 'reversed'
+    if (typeof key === 'object') {
+      const path = doCalculatePath(key, object);
+      if (path) {
+        return [value, ...path];
+      }
+    }
+  }
+  return undefined;
+};
+
+export const getFromPath = (
+  root: any,
+  path: Array<string>
+): any | GetPathError => {
+  const element = doGetFromPath(root, path);
+  if (!element) {
+    return {
+      id: 'getPathError',
+      root: root,
+      path: path,
+    };
+  }
+  return element;
+};
+
+const doGetFromPath = (root: any, path: Array<string>): any => {
+  if (path.length === 0) {
+    return root;
+  }
+  const [pathElement, ...rest] = path;
+  if (root instanceof Map) {
+    if (root.has(pathElement)) {
+      return getFromPath(root.get(pathElement), rest);
+    }
+    // must be a reverse map
+    const element = Array.from(root.entries()).reduce((acc, [key, value]) => {
+      if (value === pathElement) {
+        return key;
+      }
+      return acc;
+    }, undefined);
+    return getFromPath(element, rest);
+  }
+  return getFromPath(get(root, [pathElement]), rest);
+};
+
+export const linkElements = (
+  uiSchemaElement: EditorUISchemaElement,
+  schemaElement: SchemaElement
+): boolean => {
+  if (!uiSchemaElement.uuid) {
+    console.error('Found element without UUID', uiSchemaElement);
+    return false;
+  }
+
+  (schemaElement.linkedUISchemaElements =
+    schemaElement.linkedUISchemaElements || new Set()).add(
+    uiSchemaElement.uuid
+  );
+
+  uiSchemaElement.linkedSchemaElement = schemaElement.uuid;
+  return true;
+};
+
+export const buildAndLinkUISchema = (
+  schema: SchemaElement | undefined,
+  uiSchema: any
```

Comment:
As mentioned int he other comment this method should be split into two and shouldn't use `withCloneTrees` itself. Also the uischema type should be set appropriatly.